src/RA_Dlg_MemBookmark.h

```diff
@@ -3,42 +3,92 @@
 #pragma once
 
 #include "ra_fwd.h"
+#include "ra_type_traits.h"
 
 class MemBookmark
 {
-public:
-    void SetDescription(const std::wstring& string) { m_sDescription = string; }
-    void SetAddress(unsigned int nVal) noexcept { m_nAddress = nVal; }
-    void SetType(unsigned int nVal) noexcept { m_nType = nVal; }
-    void SetValue(unsigned int nVal) noexcept { m_sValue = nVal; }
-    void SetPrevious(unsigned int nVal) noexcept { m_sPrevious = nVal; }
-    void IncreaseCount() noexcept { m_nCount++; }
-    void ResetCount() noexcept { m_nCount = 0; }
-
-    void SetFrozen(bool b) noexcept { m_bFrozen = b; }
-    void SetDecimal(bool b) noexcept { m_bDecimal = b; }
-
-    inline const std::wstring& Description() const noexcept { return m_sDescription; }
-    unsigned int Address() const noexcept { return m_nAddress; }
-    unsigned int Type() const noexcept { return m_nType; }
-    unsigned int Value() const noexcept { return m_sValue; }
-    unsigned int Previous() const noexcept { return m_sPrevious; }
-    unsigned int Count() const noexcept { return m_nCount; }
-
-    bool Frozen() const noexcept { return m_bFrozen; }
-    bool Decimal() const noexcept { return m_bDecimal; }
+public: // Too much trouble making this a literal type
+    MemBookmark() noexcept = default;
+    ~MemBookmark() noexcept = default;
+    // Copying deleted to enforce uniqueness and speed
+    MemBookmark(const MemBookmark&) = delete;
```

Comment:
This doesn't mean that there can't be two bookmarks with the same address, it just means that it can't be copied in the same execution (uniqueness).
Think if it like `unordered_multimap `where stuff is sorted into buckets if you want them sorted but doesn't sort automatically.